convert2rhel/pkgmanager/handlers/yum.py

```diff
@@ -0,0 +1,272 @@
+        transaction: downgrade, reinstall and downgrade. The downgrade only
+        will be executed in case of the the reinstall step raises the
+        `ReinstallInstallError`.
+        """
+        original_os_pkgs = get_system_packages_for_replacement()
+        self._set_up_base()
+        self._enable_repos()
+
+        loggerinst.info("Performing update, reinstall and downgrade of the %s packages ..." % system_info.name)
+        for pkg in original_os_pkgs:
+            self._base.update(name=pkg)
+            try:
+                self._base.reinstall(name=pkg)
+            except pkgmanager.Errors.ReinstallInstallError:
+                self._base.downgrade(name=pkg)
+            except pkgmanager.Errors.ReinstallRemoveError:
+                loggerinst.info("Can't remove pkg: %s", pkg)
+
+    def _resolve_dependencies(self, test_transaction):
+        """Try to resolve the transaction dependencies.
+
+        This method will try to resolve the dependencies of the packages that
+        are held in the transaction, we might need to call this internal method
+        multiple times, that's why it's separated from the rest of the
+        `_perform_operations()` internal method.
+
+        .. notes::
+            For YumBase().resolveDeps() the "exceptions" that can arise from this
+            actually returns in the form of a tuple (int, str | list[str]),
+            meaning that the error codes will have a different meaning and
+            message depending on the number.
+            For example:
+              0. Transaction finished successfully, but it was empty.
+              1. Any general error that happened during the transaction
+              2. Transaction finished successfully, being able to process any
+              package it had in the transaction set.
+
+            We actually need to loop through this for some time to remove all the
+            packages that are causing dependencies problems. This is stated in
+            the yum source code for the `resolveDeps` function, that if any
+            errors are "thrown" for the user, you will need to loop until the
+            point you don't have any more errors.
+
+        :return: A boolean indicating if it was successful or not.
+        :rtype: bool
+        """
+        loggerinst.info("Resolving yum dependencies for the transaction.\n")
+        ret_code, msg = self._base.resolveDeps()
+
+        # For the return code 1, yum can output two kinds of error, one being
+        # that it reached the limit for depsolving, and the actual dependencies
+        # that caused an problem.
+        if ret_code == 1:
+            # We want to fail earlier in the process, so let's check for this
+            # only when testing the transaction.
+            if "Depsolving loop limit reached" not in msg and test_transaction:
+                _resolve_yum_problematic_dependencies(msg)
+                return False
+        return True
+
```

Comment:
```suggestion
        return True
        loggerinst.info("Resolved yum dependencies.")
```

Reply:
Updated the suggestion to have the loggerinst before the `return`.